### PR TITLE
Allow HTML Reader to load from string

### PR DESCRIPTION
**Pull request by gnat42**
We often want to export a table as an excel sheet. The system renders the html and it seems like a waste of time to write it to the file system to use the reader. This allows us to render the html and then just pass it to a reader

This is:

```
- [ ] a bugfix
- [x] a new feature
```

Checklist:

- [x] Changes are covered by unit tests
- [x] Code style is respected
- [x] Commit message explains **why** the change is made (see https://github.com/erlang/otp/wiki/Writing-good-commit-messages)
- [x] CHANGELOG.md contains a short summary of the change
- [x] Documentation is updated as necessary

### Why this change is needed?
Just allows Reader\Html to accept html content as a string instead of just a file.

### CODE REVIEW

**Comment by gnat42** [comment]
I can add the rest of the checklist items if this is something that you think you would merge

**Comment by PowerKiKi** [comment]
It sounds like it could be useful, but to be merged the checklist will have to be completed and the code should not be duplicated, instead create a common private method used by both `load` and `loadFromString`. Most likely that new method would receive a `DOMDocument`. Also be sure to make use of `$this->securityScanner`

**Comment by gnat42** [comment]
@PowerKiKi  Sounds good, I just didn't want to go to a whole lot of trouble if the basic idea wouldn't be accepted. I've rebased and re-worked the PR to include all checklist items. Let me know if there are any issues that need tweaking.

**Comment by codysnider** [comment]
@gnat42 Let me know if you need help with this. I was planning to put something similar in today but looks like you are 90% there.

**Comment by gnat42** [comment]
What's missing? I added tests and docs and added to the changelog?

File: src/PhpSpreadsheet/Reader/Html.php [hunk 1]
```diff
@@ -606,14 +600,56 @@ public function loadIntoExisting($pFilename, Spreadsheet $spreadsheet)
...
+    {
+        //    Create a new DOM object
+        $dom = new DOMDocument();
+        //    Reload the HTML file into the DOM object
+        $loaded = $dom->loadHTML(mb_convert_encoding($this->securityScanner->scan($content), 'HTML-ENTITIES', 'UTF-8'));
+        if ($loaded === false) {
+            throw new Exception('Failed to load content as a DOM Document');
+        }
+
+        return $this->loadDom($dom, new Spreadsheet());
```
Comment (codysnider) [inline src/PhpSpreadsheet/Reader/Html.php]:
```suggestion
        return $this->loadDocument($dom, new Spreadsheet());
```

**Comment by codysnider** [comment]
As far as I can tell, just that (picked up by scrutinizer).

**Comment by gnat42** [comment]
So I'm a bit confused as to how the code coverage decreased. I have a test that tests loadFromString and it passes, but the code coverage makes it out as if loadFromString isn't tested at all.

**Comment by codysnider** [comment]
You may want to assign @PowerKiKi as both an assignee and a reviewer. I'm not a maintainer and haven't the means to approve or deny this PR.

**Comment by gnat42** [comment]
Hey @codysnider I'm not sure I'm able to assign anyone. Hopefully @PowerKiKi sees and assigns himself.

**Comment by PowerKiKi** [comment]
Thanks ! it has been merged as 95c8bb9918b43a1f088cab54c4684e1ef40caaaa (with minor modifications)

**Pull request closed without merging**